octobot_trading/exchanges/data/exchange_personal_data.py

```diff
@@ -118,26 +117,55 @@ async def handle_order_update_from_raw(self, symbol, order_id, raw_order, should
     async def handle_order_instance_update(self, order, should_notify: bool = True):
         try:
             changed: bool = self.orders_manager.upsert_order_instance(order)
-            if should_notify:
-                await get_chan(ORDERS_CHANNEL, self.exchange_manager.id).get_internal_producer() \
-                    .send(cryptocurrency=self.exchange_manager.exchange.get_pair_cryptocurrency(order.symbol),
-                          symbol=order.symbol,
-                          order=order.to_dict(),
-                          is_from_bot=True,
-                          is_closed=False,
-                          is_updated=changed)
+
+            if changed:
+                if await self._handle_order_post_update(order):
+                    # when the order is removed from orders_manager
+                    return changed
+
+                if should_notify:
+                    await self._handle_order_update_notification(order, changed)
+
             return changed
         except Exception as e:
             self.logger.exception(e, True, f"Failed to update order instance : {e}")
             return False
 
-    async def handle_closed_order_update(self, symbol, order_id, raw_order, should_notify: bool = True) -> bool:
+    async def _handle_order_update_notification(self, order, changed):
+        """
+        Notify Orders channel for Order update
+        :param order: the updated order
+        :param changed: True if the order was updated
+        """
+        await get_chan(ORDERS_CHANNEL, self.exchange_manager.id).get_internal_producer() \
+            .send(cryptocurrency=self.exchange_manager.exchange.get_pair_cryptocurrency(order.symbol),
+                  symbol=order.symbol,
+                  order=order.to_dict(),
+                  is_from_bot=order.is_from_this_octobot,
+                  is_closed=order.is_open(),
+                  is_updated=changed)
+
+    async def _handle_order_post_update(self, order) -> bool:
+        """
+        Handle order status updates (OrderStatus.FILLED, OrderStatus.CLOSED)
+        :param order: the updated order
+        :return True if an action was performed
+        """
+        # TODO manage OrderStatus.PARTIALLY_FILLED
+        if order.is_closed():
+            return await self.handle_closed_order_update(order.symbol, order.order_id, order.to_dict(),
+                                                         should_notify=True)
```

Comment:
should return False by default instead of None to be more clear

Reply:
Yes  :+1: